File: src/main/java/com/dozycoffee/wms/global/error/DomainValidator.java
package com.dozycoffee.wms.global.error;

public final class DomainValidator {

    private DomainValidator() {
    }

    public static <T> T requireNonNull(T value, ErrorCode errorCode) {
        if (value == null) {
            throw new InvalidDomainValueException(errorCode);
        }
        return value;
    }
}

File: src/main/kotlin/com/dozycoffee/wms/global/error/DomainValidator.kt
package com.dozycoffee.wms.global.error

object DomainValidator {

    @JvmStatic
    fun <T : Any> requireNonNull(value: T?, errorCode: ErrorCode): T {
        if (value == null) {
            throw InvalidDomainValueException(errorCode)
        }
        return value
    }
}
